admin/spam/views.py

```diff
@@ -1,47 +1,149 @@
 from django.shortcuts import render
-from django.core.paginator import Paginator, EmptyPage, PageNotAnInteger
 from django.contrib.auth.decorators import login_required
+from django.views.generic import FormView, ListView
+from django.utils.decorators import method_decorator
+from django.core.urlresolvers import reverse
+from django.views.defaults import page_not_found
 
 from modularodm import Q
 from website.project.model import Comment
 
 from .serializers import serialize_comment
+from .forms import ConfirmForm
 
 
-def get_spam_list():
-    query = (
-        Q('reports', 'ne', {}) &
-        Q('reports', 'ne', None)
-    )
-    return Comment.find(query).sort('date_created')
+def key_order_comments(comment):
```

Comment:
This appears to be the O(n^2) sorting mechanism mentioned in the PR description, but I don't see it being used anywhere. [This line](https://github.com/CenterForOpenScience/osf.io/pull/5045/files#diff-43b817f0cafd09bc0da4a5351442a2daR31) appears to be what's specifying the ordering.